src/rt/cover.d

```diff
@@ -149,10 +149,75 @@ uint parseNum(const(char)[] s)
 T min(T)(T a, T b) { return a < b ? a : b; }
 T max(T)(T a, T b) { return b < a ? a : b; }
 
+version (unittest) __gshared bool inUnittest;
+
+// Parse a string looking like a boolean and assign the result to the global
+// merge variable.
+string parseMergeOpt(string opt) @nogc nothrow
+{
+    switch (opt)
+    {
+        case "no", "No", "NO", "n", "N", "0":
+            merge = false;
+            return "0";
+        case "", "yes", "Yes", "YES", "y", "Y", "1":
+            merge = true;
+            return "1";
+        default:
+            version (unittest) if (inUnittest) return "err";
+            fprintf(stderr, "DRT Error: Runtime option 'covmerge' " ~
+                    "should be 1/y/yes or 0/n/no, got '%.*s'.\n",
+                    cast(int)opt.length, opt.ptr);
+            return "err";
+    }
+}
+
+unittest
+{
+    inUnittest = true;
+    scope (exit) inUnittest = false;
+
+    bool old_merge = merge;
+    scope (exit) merge = old_merge;
+
+    merge = true;
+    assert(parseMergeOpt("0") == "0" && !merge);
+    assert(parseMergeOpt("1") == "1" && merge);
+    assert(parseMergeOpt("n") == "0" && !merge);
+    assert(parseMergeOpt("y") == "1" && merge);
+    assert(parseMergeOpt("N") == "0" && !merge);
+    assert(parseMergeOpt("Y") == "1" && merge);
+    assert(parseMergeOpt("No") == "0" && !merge);
+    assert(parseMergeOpt("Yes") == "1" && merge);
+    assert(parseMergeOpt("NO") == "0" && !merge);
+    assert(parseMergeOpt("YES") == "1" && merge);
+    assert(parseMergeOpt("no") == "0" && !merge);
+    assert(parseMergeOpt("yes") == "1" && merge);
+    assert(parseMergeOpt("no") == "0" && !merge);
+    assert(parseMergeOpt("") == "1" && merge);
+    merge = false;
+    assert(parseMergeOpt("yEs") == "err" && !merge);
+    merge = true;
+    assert(parseMergeOpt("nO") == "err" && merge);
+    merge = false;
+    assert(parseMergeOpt("10") == "err" && !merge);
+    merge = true;
+    assert(parseMergeOpt("01") == "err" && merge);
+    assert(parseMergeOpt("xxx") == "err" && merge);
+    assert(parseMergeOpt("invalid") == "err" && merge);
+}
+
 shared static ~this()
 {
     if (!gdata.length) return;
 
+    // Parse options taken from the standard runtime options passing via
+    // --DRT-xxx / DRT_XXX.
+    import rt.config: rt_configOption;
+    rt_configOption("covmerge", (string opt) => parseMergeOpt(opt), false);
+    rt_configOption("covsrcpath", (string opt) => srcpath = opt, false);
+    rt_configOption("covdstpath", (string opt) => dstpath = opt, false);
```

Comment:
The format is a lot different from `DRT-gcopt`.
How about something more consistent?
```sh
-DRT-coverage=help

merge:0|1 - Set whether to merge data with listings from previous runs.
srcpath:path - Set path to where source files are located.
dstpath:path - Set path to where listing files are to be written.
```
So for example `-DRT-coverage='merge:1 srcpath:C:\foo\bar dstpath:outdir'` would be used for all configs.
Also an opportunity to the existing parsing code from `gc/config.d` to `core/internal/drt_config.d` or so.

Reply:
I tried to start like that, but then I found it was quite difficult to abstract out the code from `gcopts` because of this *no imports of rt from gc*. Then also, if there is already mechanisms to easily add `--DTR-xxx` options, why bothering complicating stuff by adding another layer of parsing?

Reply:
To be more precise, what I mean is the code in the gc is generic to be able to parse a struct, but I don't know if I can import from the GC those templates if they are moved to the runtime (where they should belong if it will be a standard way to add options). If this can be done, I would need some help because I'm not familiar with it.

Why is this restriction about importing `rt.*` modules inside the GC again?

Reply:
But I still think is silly to have 2 layers of parsing in options, if we go generic I think it might be better to head for something like `--DRT-gc-disable=1 --DRT-cov-merge=0` and maybe add a `--DRT-help` to get all the options.

I don't think this PR is the place to go for that refactoring thought.